app/services/storage_root_migration_service.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+# Migrates Complete Moab records to a new Moab Storate Root.
+class StorageRootMigrationService
+  def initialize(from_name, to_name)
+    @from_name = from_name
+    @to_name = to_name
+  end
+
+  # @return [Array<String>] druids of migrated moabs
+  def migrate
+    druids = []
+    from_root.complete_moabs.find_each do |complete_moab|
+      migrate_moab(complete_moab)
+      druids << complete_moab.preserved_object.druid
+    end
+    druids
+  end
+
+  private
+
+  def from_root
+    @from_root ||= MoabStorageRoot.find_by(name: @from_name)
```

Comment:
```suggestion
    @from_root ||= MoabStorageRoot.find_by!(name: @from_name)
```

seems better to fail ASAP, `find_by!` will `raise` if the row doesn't exist.